Sources/ESRefreshComponent.swift

```diff
@@ -29,60 +29,70 @@ import UIKit
 public typealias ESRefreshHandler = (() -> ())
 
 open class ESRefreshComponent: UIView {
-    
+
     open weak var scrollView: UIScrollView?
-    
+
     /// @param handler Refresh callback method
     open var handler: ESRefreshHandler?
-    
+
     /// @param animator Animated view refresh controls, custom must comply with the following two protocol
     open var animator: (ESRefreshProtocol & ESRefreshAnimatorProtocol)!
-    
+
     /// @param refreshing or not
     fileprivate var _isRefreshing = false
     open var isRefreshing: Bool {
         get {
             return self._isRefreshing
         }
     }
-    
+
     /// @param auto refreshing or not
     fileprivate var _isAutoRefreshing = false
     open var isAutoRefreshing: Bool {
         get {
             return self._isAutoRefreshing
         }
     }
-    
+
     /// @param tag observing
     fileprivate var isObservingScrollView = false
     fileprivate var isIgnoreObserving = false
+    fileprivate var offsetObservation: NSKeyValueObservation? {
+        willSet {
+            offsetObservation?.invalidate()
```

Comment:
Do we need these invalidations? I thought part of the pitch of block-based KVO was that you got this for free when the ref count went to 0

Reply:
No we you're right don't just did it for safety - will remove.